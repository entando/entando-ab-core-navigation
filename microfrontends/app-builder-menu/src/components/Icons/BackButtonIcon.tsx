import { COLORS } from '../theme';

interface Props {
  width?: number;
  height?: number;
  fill?: string;
}

function BackButtonIcon(props: Props): JSX.Element {
  const { width = 24, height = 24, fill = COLORS.blue } = props;
  return (
    <svg
      xmlns="http://www.w3.org/2000/svg"
      width={width}
      height={height}
      viewBox="0 0 24 24"
      fill="none"
      stroke={fill}
      strokeWidth="2"
      strokeLinecap="round"
      strokeLinejoin="round"
    >
      <circle cx="12" cy="12" r="10"></circle>
      <polyline points="12 8 8 12 12 16"></polyline>
      <line x1="16" y1="12" x2="8" y2="12"></line>
    </svg>
  );
}

export default BackButtonIcon;
