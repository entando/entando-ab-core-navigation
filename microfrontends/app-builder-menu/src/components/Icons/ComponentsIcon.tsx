interface Props {
  width?: number;
  height?: number;
  fill?: string;
}

function ComponentsIcon(props: Props): JSX.Element {
  const { width = 20, height = 20, fill = '#72767B' } = props;
  return (
    <svg
      width={width}
      height={height}
      viewBox={`0 0 ${width} ${height}`}
      fill="none"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        fillRule="evenodd"
        clipRule="evenodd"
        d="M20 8.47307C20 8.70305 19.8134 8.88951 19.5833 8.88951H19.1667V14.442H19.5833C19.8134 14.442 20 14.6284 20 14.8584V17.357C20 17.587 19.8134 17.7734 19.5833 17.7734H17.0833C16.8532 17.7734 16.6667 17.587 16.6667 17.357V16.9406H8.88889V17.357C8.88889 17.587 8.70233 17.7734 8.47222 17.7734H5.97222C5.74212 17.7734 5.55556 17.587 5.55556 17.357V14.8584C5.55556 14.6284 5.74212 14.442 5.97222 14.442H6.38889V13.6091H3.33333V14.0255C3.33333 14.2555 3.14677 14.442 2.91667 14.442H0.416667C0.186563 14.442 0 14.2555 0 14.0255V11.5269C0 11.2969 0.186563 11.1105 0.416667 11.1105H0.833333V5.55804H0.416667C0.186563 5.55804 0 5.37158 0 5.1416V2.643C0 2.41302 0.186563 2.22656 0.416667 2.22656H2.91667C3.14677 2.22656 3.33333 2.41302 3.33333 2.643V3.05943H11.1111V2.643C11.1111 2.41302 11.2977 2.22656 11.5278 2.22656H14.0278C14.2579 2.22656 14.4444 2.41302 14.4444 2.643V5.1416C14.4444 5.37158 14.2579 5.55804 14.0278 5.55804H13.6111V6.3909H16.6667V5.97447C16.6667 5.74449 16.8532 5.55804 17.0833 5.55804H19.5833C19.8134 5.55804 20 5.74449 20 5.97447V8.47307ZM13.3333 3.33703H12.2222V4.44752H13.3333V3.33703ZM12.2222 12.221H13.3333V13.3315H12.2222V12.221ZM1.1111 13.3315H2.22221V12.221H1.1111V13.3315ZM2.22221 4.44752H1.1111V3.33703H2.22221V4.44752ZM3.33333 11.5269V11.9434H11.1111V11.5269C11.1111 11.297 11.2977 11.1105 11.5278 11.1105H11.9444V5.55803H11.5278C11.2977 5.55803 11.1111 5.37157 11.1111 5.14159V4.72516H3.33333V5.14159C3.33333 5.37157 3.14677 5.55803 2.91667 5.55803H2.5V11.1105H2.91667C3.14677 11.1105 3.33333 11.297 3.33333 11.5269ZM7.77776 16.663H6.66665V15.5525H7.77776V16.663ZM17.0833 14.442H17.5V8.88951H17.0833C16.8532 8.88951 16.6667 8.70305 16.6667 8.47308V8.05664H13.6111V11.1105H14.0278C14.2579 11.1105 14.4444 11.297 14.4444 11.5269V14.0255C14.4444 14.2555 14.2579 14.442 14.0278 14.442H11.5278C11.2977 14.442 11.1111 14.2555 11.1111 14.0255V13.6091H8.05556V14.442H8.47222C8.70233 14.442 8.88889 14.6284 8.88889 14.8584V15.2749H16.6667V14.8584C16.6667 14.6284 16.8532 14.442 17.0833 14.442ZM18.8889 16.663H17.7778V15.5525H18.8889V16.663ZM17.7778 7.77901H18.8889V6.66852H17.7778V7.77901Z"
        fill="black"
      />
      <mask
        id="mask0_277_1610"
        maskUnits="userSpaceOnUse"
        x="0"
        y="2"
        width="20"
        height="16"
      >
        <path
          fillRule="evenodd"
          clipRule="evenodd"
          d="M20 8.47307C20 8.70305 19.8134 8.88951 19.5833 8.88951H19.1667V14.442H19.5833C19.8134 14.442 20 14.6284 20 14.8584V17.357C20 17.587 19.8134 17.7734 19.5833 17.7734H17.0833C16.8532 17.7734 16.6667 17.587 16.6667 17.357V16.9406H8.88889V17.357C8.88889 17.587 8.70233 17.7734 8.47222 17.7734H5.97222C5.74212 17.7734 5.55556 17.587 5.55556 17.357V14.8584C5.55556 14.6284 5.74212 14.442 5.97222 14.442H6.38889V13.6091H3.33333V14.0255C3.33333 14.2555 3.14677 14.442 2.91667 14.442H0.416667C0.186563 14.442 0 14.2555 0 14.0255V11.5269C0 11.2969 0.186563 11.1105 0.416667 11.1105H0.833333V5.55804H0.416667C0.186563 5.55804 0 5.37158 0 5.1416V2.643C0 2.41302 0.186563 2.22656 0.416667 2.22656H2.91667C3.14677 2.22656 3.33333 2.41302 3.33333 2.643V3.05943H11.1111V2.643C11.1111 2.41302 11.2977 2.22656 11.5278 2.22656H14.0278C14.2579 2.22656 14.4444 2.41302 14.4444 2.643V5.1416C14.4444 5.37158 14.2579 5.55804 14.0278 5.55804H13.6111V6.3909H16.6667V5.97447C16.6667 5.74449 16.8532 5.55804 17.0833 5.55804H19.5833C19.8134 5.55804 20 5.74449 20 5.97447V8.47307ZM13.3333 3.33703H12.2222V4.44752H13.3333V3.33703ZM12.2222 12.221H13.3333V13.3315H12.2222V12.221ZM1.1111 13.3315H2.22221V12.221H1.1111V13.3315ZM2.22221 4.44752H1.1111V3.33703H2.22221V4.44752ZM3.33333 11.5269V11.9434H11.1111V11.5269C11.1111 11.297 11.2977 11.1105 11.5278 11.1105H11.9444V5.55803H11.5278C11.2977 5.55803 11.1111 5.37157 11.1111 5.14159V4.72516H3.33333V5.14159C3.33333 5.37157 3.14677 5.55803 2.91667 5.55803H2.5V11.1105H2.91667C3.14677 11.1105 3.33333 11.297 3.33333 11.5269ZM7.77776 16.663H6.66665V15.5525H7.77776V16.663ZM17.0833 14.442H17.5V8.88951H17.0833C16.8532 8.88951 16.6667 8.70305 16.6667 8.47308V8.05664H13.6111V11.1105H14.0278C14.2579 11.1105 14.4444 11.297 14.4444 11.5269V14.0255C14.4444 14.2555 14.2579 14.442 14.0278 14.442H11.5278C11.2977 14.442 11.1111 14.2555 11.1111 14.0255V13.6091H8.05556V14.442H8.47222C8.70233 14.442 8.88889 14.6284 8.88889 14.8584V15.2749H16.6667V14.8584C16.6667 14.6284 16.8532 14.442 17.0833 14.442ZM18.8889 16.663H17.7778V15.5525H18.8889V16.663ZM17.7778 7.77901H18.8889V6.66852H17.7778V7.77901Z"
          fill="white"
        />
      </mask>
      <g mask="url(#mask0_277_1610)">
        <path
          fillRule="evenodd"
          clipRule="evenodd"
          d="M0 0H20V20H0V0Z"
          fill={fill}
        />
      </g>
    </svg>
  );
}

export default ComponentsIcon;
