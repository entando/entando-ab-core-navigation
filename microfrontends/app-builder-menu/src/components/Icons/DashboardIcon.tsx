import { COLORS } from '../theme';

interface Props {
  width?: number;
  height?: number;
  fill?: string;
}

function DashboardIcon(props: Props): JSX.Element {
  const { width = 20, height = 20, fill = COLORS.inactive } = props;
  return (
    <svg
      width={width}
      height={height}
      viewBox={`0 0 ${width} ${height}`}
      fill="none"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        fillRule="evenodd"
        clipRule="evenodd"
        d="M1.875 1.25H18.125C19.1602 1.25 20 2.08984 20 3.125V16.875C20 17.9102 19.1602 18.75 18.125 18.75H1.875C0.839844 18.75 0 17.9102 0 16.875V3.125C0 2.08984 0.839844 1.25 1.875 1.25ZM17.8906 16.875C18.0195 16.875 18.125 16.7695 18.125 16.6406V7.5H1.875V16.6406C1.875 16.7695 1.98047 16.875 2.10938 16.875H17.8906Z"
        fill="black"
      />
      <mask
        id="mask0_277_1709"
        maskUnits="userSpaceOnUse"
        x="0"
        y="1"
        width="20"
        height="18"
      >
        <path
          fillRule="evenodd"
          clipRule="evenodd"
          d="M1.875 1.25H18.125C19.1602 1.25 20 2.08984 20 3.125V16.875C20 17.9102 19.1602 18.75 18.125 18.75H1.875C0.839844 18.75 0 17.9102 0 16.875V3.125C0 2.08984 0.839844 1.25 1.875 1.25ZM17.8906 16.875C18.0195 16.875 18.125 16.7695 18.125 16.6406V7.5H1.875V16.6406C1.875 16.7695 1.98047 16.875 2.10938 16.875H17.8906Z"
          fill="white"
        />
      </mask>
      <g mask="url(#mask0_277_1709)">
        <path
          fillRule="evenodd"
          clipRule="evenodd"
          d="M0 0H20V20H0V0Z"
          fill={fill}
        />
      </g>
    </svg>
  );
}

export default DashboardIcon;
