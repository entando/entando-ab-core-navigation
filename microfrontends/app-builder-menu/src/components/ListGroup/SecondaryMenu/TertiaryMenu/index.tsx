import styled from 'styled-components';

const StyledSecondaryTertiaryMenu = styled.div<{ isVisible: boolean }>`
  background: #292e34;
  border: 1px solid #292e34;
  border-bottom: none;
  border-top: none;
  bottom: 0;
  display: block;
  left: 400px;
  overflow-x: hidden;
  overflow-y: auto;
  position: fixed;
  top: 0px;
  opacity: 0;
  visibility: hidden;
  ${({ isVisible }) => (isVisible ? `opacity: 1; visibility: visible;` : ``)}
  width: 200px;
  z-index: 1030;
`;

const StyledSecondaryListGroup = styled.ul`
  margin-top: 25px;
  padding-left: 22px;
  padding-right: 0px;
`;

interface Props {
  isOpen: boolean;
  children?: React.ReactNode;
}

export default function TertiaryMenu(props: Props): JSX.Element {
  const { isOpen, children } = props;
  return (
    <StyledSecondaryTertiaryMenu isVisible={isOpen}>
      <StyledSecondaryListGroup>{children}</StyledSecondaryListGroup>
    </StyledSecondaryTertiaryMenu>
  );
}
