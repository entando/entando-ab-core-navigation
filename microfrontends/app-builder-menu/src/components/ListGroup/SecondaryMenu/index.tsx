import { useContext } from 'react';
import styled from 'styled-components';
import BackButtonIcon from '../../Icons/BackButtonIcon';
import MenuUIContext from '../../MenuUIContext';
import { COLORS } from '../../theme';

const StyledSecondaryNavMenu = styled.div<{ isVisible: boolean }>`
  background: ${COLORS.active};
  border: 1px solid ${COLORS.primary};
  border-bottom: none;
  border-top: none;
  bottom: 0;
  display: block;
  left: 200px;
  overflow-x: hidden;
  overflow-y: auto;
  position: fixed;
  top: 0px;
  opacity: 0;
  visibility: hidden;
  ${({ isVisible }) => (isVisible ? `opacity: 1; visibility: visible;` : ``)}
  width: 200px;
  z-index: 1030;
`;

const StyledHeader = styled.div`
  display: flex;
  align-items: center;
  color: ${COLORS.text};
  font-weight: 600;
  margin-top: 19px;
  margin-left: 28px;
`;

const StyledTitle = styled.span`
  margin-left: 14px;
`;

const StyledSecondaryListGroup = styled.ul`
  margin-top: 23px;
  padding-left: 24px;
  padding-right: 0px;
`;

const StyledBackButton = styled.div`
  display: flex;
  align-items: center;
  cursor: pointer;
`;

interface Props {
  isOpen: boolean;
  title: string;
  children?: React.ReactNode;
}

export default function SecondaryMenu(props: Props): JSX.Element {
  const { isOpen, title, children } = props;

  const { setSecondaryMenuOpen, setTertiaryMenuOpen } =
    useContext(MenuUIContext);

  const handleBack = () => {
    setSecondaryMenuOpen(false);
    setTertiaryMenuOpen(false);
  };

  return (
    <StyledSecondaryNavMenu isVisible={isOpen}>
      <StyledHeader>
        <StyledBackButton onClick={handleBack}>
          <BackButtonIcon />
        </StyledBackButton>
        <StyledTitle>{title}</StyledTitle>
      </StyledHeader>
      <StyledSecondaryListGroup>{children}</StyledSecondaryListGroup>
    </StyledSecondaryNavMenu>
  );
}
