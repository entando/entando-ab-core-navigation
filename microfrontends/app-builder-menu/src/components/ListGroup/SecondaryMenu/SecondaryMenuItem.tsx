import styled from 'styled-components';
import RightArrowIcon from '../../Icons/RightArrowIcon';
import TertiaryMenu from './TertiaryMenu';

const StyledLink = styled.a<{ isActive: boolean }>`
  width: 100%;
  background-color: transparent;
  background-color: ${({ isActive }) => (isActive ? '#4D5257' : 'transparent')};
  color: ${({ isActive }) => (isActive ? '#fff' : '#d1d1d1')};
  cursor: pointer;
  display: block;
  font-size: 14px;
  font-weight: ${({ isActive }) => (isActive ? '600' : '400')};
  line-height: 26px;
  padding: 5px 10px 5px 8px;
  position: relative;
  white-space: nowrap;
  text-decoration: none;
  display: flex;
  margin-top: 4px;

  &:hover {
    font-weight: 600;
    color: #fff;
    background-color: #4d5257;
  }
`;

const StyledLabel = styled.span`
  display: block;
  flex: 1;
  max-width: none;
  overflow: hidden;
  text-overflow: ellipsis;
`;

const StyledSecondaryMenuItem = styled.li`
  display: flex;
`;

interface Props {
  isActive: boolean;
  hasChildren: boolean;
}

export default function SecondaryMenuItem(props: Props): JSX.Element {
  const { isActive, hasChildren } = props;
  return (
    <StyledSecondaryMenuItem>
      <StyledLink isActive={isActive}>
        <StyledLabel>Dashboard</StyledLabel>
        {hasChildren && <RightArrowIcon />}
      </StyledLink>
      <TertiaryMenu isOpen />
    </StyledSecondaryMenuItem>
  );
}
