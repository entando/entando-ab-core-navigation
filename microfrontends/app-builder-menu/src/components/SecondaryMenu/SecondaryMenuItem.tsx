import { SyntheticEvent, useContext } from 'react';
import styled from 'styled-components';
import { RightArrowIcon } from './../Icons/RightArrowIcon';
import { MenuUIContext } from './../MenuUIContext';
import { TertiaryMenu } from '../TertiaryMenu/TertiaryMenu';
import { COLORS } from './../theme';

const StyledLink = styled.a<{ isActive: boolean }>`
  width: 100%;
  background-color: transparent;
  background-color: ${({ isActive }) =>
    isActive ? COLORS.secondary : 'transparent'};
  color: ${({ isActive }) => (isActive ? COLORS.white : COLORS.text)};
  cursor: pointer;
  display: block;
  font-size: 14px;
  font-family: 'Open Sans', sans-serif;
  font-weight: ${({ isActive }) => (isActive ? '600' : '400')};
  line-height: 26px;
  padding: 5px 10px 5px 8px;
  position: relative;
  white-space: nowrap;
  text-decoration: none;
  display: flex;
  margin-top: 4px;

  &:hover {
    font-weight: 600;
    color: ${COLORS.white};
    background-color: ${COLORS.secondary};
  }
`;

const StyledLabel = styled.span`
  display: block;
  flex: 1;
  max-width: none;
  overflow: hidden;
  text-overflow: ellipsis;
`;

const StyledSecondaryMenuItem = styled.li`
  display: flex;
`;

interface Props {
  id: string;
  dataId: string;
  children?: React.ReactNode;
  label: string;
  onClick?: () => void;
  href?: string;
  className?: string;
}

export function SecondaryMenuItem(props: Props): JSX.Element {
  const {
    id,
    dataId,
    children,
    label,
    onClick,
    href,
    className = '',
  } = props;

  const {
    activeSecondaryMenuItemId,
    setActiveSecondaryMenuItemId,
    tertiaryMenuOpen,
    setTertiaryMenuOpen
  } = useContext(MenuUIContext);

  const hasChildren = !!children;
  const isActive = activeSecondaryMenuItemId === id;

  const handleClick = (e: SyntheticEvent): void => {
    e.stopPropagation()

    setActiveSecondaryMenuItemId(id);
    if (hasChildren) {
      setTertiaryMenuOpen(true);
    } else {
      onClick && onClick();
    }
  };

  return (
    <>
      <StyledSecondaryMenuItem
        onClick={handleClick}
        className={className}
        data-id={dataId}
      >
        <StyledLink isActive={isActive} href={href}>
          <StyledLabel>{label}</StyledLabel>
          {hasChildren && <RightArrowIcon />}
        </StyledLink>
        {hasChildren && isActive && tertiaryMenuOpen && (
          <TertiaryMenu isOpen>{children}</TertiaryMenu>
        )}
      </StyledSecondaryMenuItem>
    </>
  );
}
