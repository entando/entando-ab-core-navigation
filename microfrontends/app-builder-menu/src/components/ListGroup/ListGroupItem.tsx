import { useContext } from 'react';
import styled from 'styled-components';
import { RightArrowIcon } from '../Icons/RightArrowIcon';
import { MenuUIContext } from '../MenuUIContext';
import { SecondaryMenu } from '../SecondaryMenu/SecondaryMenu';
import { COLORS } from '../theme';

interface ListGroupItemProps {
  fixBottom?: boolean;
  isActive?: boolean;
}

const StyledListGroupItem = styled.li<ListGroupItemProps>`
  position: relative;
  display: block;
  padding: 10px 15px;
  margin-bottom: -1px;
  border: 1px solid ${COLORS.border};
  border-left: 0;
  border-right: 0;
  background-color: transparent;
  padding: 0;
  color: ${({ isActive }) => (isActive ? COLORS.white : '')};
  ${({ fixBottom }) =>
    fixBottom
      ? `
  position: fixed;
  bottom: 64px;
  left: 0;
  `
      : ''}

  &:first-child {
    border-top: 0px;
  }
`;

interface StyledLinkProps {
  isActive: boolean;
}

const StyledLink = styled.a<StyledLinkProps>`
  width: 215px;
  background-color: ${({ isActive }) =>
    isActive ? COLORS.active : 'transparent'};
  color: ${({ isActive }) => (isActive ? COLORS.white : COLORS.text)};
  cursor: pointer;
  display: block;
  font-size: 14px;
  font-family: 'Open Sans', sans-serif;
  font-weight: ${({ isActive }) => (isActive ? '600' : '400')};
  height: 63px;
  line-height: 26px;
  padding: 17px 20px 17px 25px;
  position: relative;
  white-space: nowrap;
  text-decoration: none;
  display: flex;
  padding-right: 0;
  ${({ isActive }) =>
    isActive
      ? `
  &:before {
    background: ${COLORS.blue};
    content: " ";
    height: 100%;
    left: 0;
    position: absolute;
    top: 0;
    width: 3px;
  }
  `
      : ''}

  &:hover {
    font-weight: 600;
    color: ${COLORS.white};
    background-color: ${COLORS.active};
    &:before {
      background: ${COLORS.blue};
      content: ' ';
      height: 100%;
      left: 0;
      position: absolute;
      top: 0;
      width: 3px;
    }

    span > svg > path {
      fill: ${COLORS.blue};
    }
    span > svg > g > path {
      fill: ${COLORS.blue};
    }
  }
`;

const StyledIcon = styled.span`
  display: inline-flex;
  align-items: center;
  color: ${COLORS.inactive};
  float: left;
  font-size: 20px;
  line-height: 26px;
  margin-right: 10px;
  text-align: center;
  width: 24px;
`;

export const StyledLabel = styled.span`
  display: flex;
  line-height: 25px;
  flex: 1;
  max-width: none;
  padding-right: 15px;
  overflow: hidden;
  text-overflow: ellipsis;
  align-items: center;
`;

interface Props {
  id: string;
  fixBottom?: boolean;
  label: string;
  renderIcon: ({ fill }: { fill?: string }) => React.ReactNode;
  children?: React.ReactNode;
  onClick?: () => void;
  href?: string;
  className?: string;
}

export function ListGroupItem(props: Props): JSX.Element {
  const {
    label,
    renderIcon,
    fixBottom,
    children,
    id,
    onClick,
    href,
    className = ''
  } = props;

  const {
    activeListGroupItemId,
    setActiveListGroupItemId,
    setSecondaryMenuOpen,
    secondaryMenuOpen,
    setActiveTertiaryMenuItemId,
    setActiveSecondaryMenuItemId
  } = useContext(MenuUIContext);

  const isActive = activeListGroupItemId === id;
  const hasChildren = !!children;

  const onClickHandler = () => {
    setActiveListGroupItemId(id);
    if (hasChildren) {
      setSecondaryMenuOpen(true);
      setActiveTertiaryMenuItemId('');
      setActiveSecondaryMenuItemId('');
    } else {
      onClick && onClick();
    }
  };

  return (
    <>
      <StyledListGroupItem
        isActive={isActive}
        fixBottom={fixBottom}
        onClick={onClickHandler}
        className={className}
      >
        <StyledLink isActive={isActive} href={href}>
          <StyledIcon>
            {renderIcon({ fill: isActive ? COLORS.blue : COLORS.inactive })}
          </StyledIcon>
          <StyledLabel>{label}</StyledLabel>
          {hasChildren && (
            <StyledIcon>
              <RightArrowIcon />{' '}
            </StyledIcon>
          )}
        </StyledLink>
      </StyledListGroupItem>
      {hasChildren && isActive && secondaryMenuOpen && (
        <SecondaryMenu isOpen title={label}>
          {children}
        </SecondaryMenu>
      )}
    </>
  );
}
