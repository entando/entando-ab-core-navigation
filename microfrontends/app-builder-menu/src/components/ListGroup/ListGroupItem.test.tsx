import React from 'react';
import renderer from 'react-test-renderer';
import { DashboardIcon } from '../Icons/DashboardIcon';
import { SecondaryMenuItem } from '../SecondaryMenu/SecondaryMenuItem';

import { ListGroupItem } from './ListGroupItem';

test('renders ListGroupItem withour children', () => {
  const tree = renderer
    .create(
      <ListGroupItem
        id="item1"
        label="Item 1"
        renderIcon={() => <DashboardIcon />}
        onClick={() => {}}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders ListGroupItem with children', () => {
  const tree = renderer
    .create(
      <ListGroupItem
        id="item1"
        label="Item 1"
        onClick={() => {}}
        renderIcon={() => <DashboardIcon />}
      >
        <SecondaryMenuItem id="child1" label="Child 1" />
      </ListGroupItem>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders ListGroupItem with href', () => {
  const tree = renderer
    .create(
      <ListGroupItem
        id="item1"
        label="Item 1"
        onClick={() => {}}
        href="http://localhost:8080"
        renderIcon={() => <DashboardIcon />}
      >
        <SecondaryMenuItem id="child1" label="Child 1" />
      </ListGroupItem>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
