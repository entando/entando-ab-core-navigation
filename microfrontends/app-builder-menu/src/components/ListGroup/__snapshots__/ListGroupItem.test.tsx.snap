// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ListGroupItem with children 1`] = `
<li
  className="sc-jqUVSM eSBrHl"
  onClick={[Function]}
>
  <a
    className="sc-kDDrLX bFWLLJ"
  >
    <span
      className="sc-iqcoie lftmcC"
    >
      <svg
        fill="none"
        height={20}
        viewBox="0 0 20 20"
        width={20}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clipRule="evenodd"
          d="M1.875 1.25H18.125C19.1602 1.25 20 2.08984 20 3.125V16.875C20 17.9102 19.1602 18.75 18.125 18.75H1.875C0.839844 18.75 0 17.9102 0 16.875V3.125C0 2.08984 0.839844 1.25 1.875 1.25ZM17.8906 16.875C18.0195 16.875 18.125 16.7695 18.125 16.6406V7.5H1.875V16.6406C1.875 16.7695 1.98047 16.875 2.10938 16.875H17.8906Z"
          fill="black"
          fillRule="evenodd"
        />
        <mask
          height="18"
          id="mask0_277_1709"
          maskUnits="userSpaceOnUse"
          width="20"
          x="0"
          y="1"
        >
          <path
            clipRule="evenodd"
            d="M1.875 1.25H18.125C19.1602 1.25 20 2.08984 20 3.125V16.875C20 17.9102 19.1602 18.75 18.125 18.75H1.875C0.839844 18.75 0 17.9102 0 16.875V3.125C0 2.08984 0.839844 1.25 1.875 1.25ZM17.8906 16.875C18.0195 16.875 18.125 16.7695 18.125 16.6406V7.5H1.875V16.6406C1.875 16.7695 1.98047 16.875 2.10938 16.875H17.8906Z"
            fill="white"
            fillRule="evenodd"
          />
        </mask>
        <g
          mask="url(#mask0_277_1709)"
        >
          <path
            clipRule="evenodd"
            d="M0 0H20V20H0V0Z"
            fill="#72767b"
            fillRule="evenodd"
          />
        </g>
      </svg>
    </span>
    <span
      className="sc-crXcEl hxLnNX"
    >
      Item 1
    </span>
    <span
      className="sc-iqcoie lftmcC"
    >
      <svg
        fill="none"
        height={20}
        stroke="#979797"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth="2"
        viewBox="0 0 20 20"
        width={20}
        xmlns="http://www.w3.org/2000/svg"
      >
        <polyline
          points="9 18 15 12 9 6"
        />
      </svg>
       
    </span>
  </a>
</li>
`;

exports[`renders ListGroupItem with href 1`] = `
<li
  className="sc-jqUVSM eSBrHl"
  onClick={[Function]}
>
  <a
    className="sc-kDDrLX bFWLLJ"
    href="http://localhost:8080"
  >
    <span
      className="sc-iqcoie lftmcC"
    >
      <svg
        fill="none"
        height={20}
        viewBox="0 0 20 20"
        width={20}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clipRule="evenodd"
          d="M1.875 1.25H18.125C19.1602 1.25 20 2.08984 20 3.125V16.875C20 17.9102 19.1602 18.75 18.125 18.75H1.875C0.839844 18.75 0 17.9102 0 16.875V3.125C0 2.08984 0.839844 1.25 1.875 1.25ZM17.8906 16.875C18.0195 16.875 18.125 16.7695 18.125 16.6406V7.5H1.875V16.6406C1.875 16.7695 1.98047 16.875 2.10938 16.875H17.8906Z"
          fill="black"
          fillRule="evenodd"
        />
        <mask
          height="18"
          id="mask0_277_1709"
          maskUnits="userSpaceOnUse"
          width="20"
          x="0"
          y="1"
        >
          <path
            clipRule="evenodd"
            d="M1.875 1.25H18.125C19.1602 1.25 20 2.08984 20 3.125V16.875C20 17.9102 19.1602 18.75 18.125 18.75H1.875C0.839844 18.75 0 17.9102 0 16.875V3.125C0 2.08984 0.839844 1.25 1.875 1.25ZM17.8906 16.875C18.0195 16.875 18.125 16.7695 18.125 16.6406V7.5H1.875V16.6406C1.875 16.7695 1.98047 16.875 2.10938 16.875H17.8906Z"
            fill="white"
            fillRule="evenodd"
          />
        </mask>
        <g
          mask="url(#mask0_277_1709)"
        >
          <path
            clipRule="evenodd"
            d="M0 0H20V20H0V0Z"
            fill="#72767b"
            fillRule="evenodd"
          />
        </g>
      </svg>
    </span>
    <span
      className="sc-crXcEl hxLnNX"
    >
      Item 1
    </span>
    <span
      className="sc-iqcoie lftmcC"
    >
      <svg
        fill="none"
        height={20}
        stroke="#979797"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth="2"
        viewBox="0 0 20 20"
        width={20}
        xmlns="http://www.w3.org/2000/svg"
      >
        <polyline
          points="9 18 15 12 9 6"
        />
      </svg>
       
    </span>
  </a>
</li>
`;

exports[`renders ListGroupItem withour children 1`] = `
<li
  className="sc-jqUVSM eSBrHl"
  onClick={[Function]}
>
  <a
    className="sc-kDDrLX bFWLLJ"
  >
    <span
      className="sc-iqcoie lftmcC"
    >
      <svg
        fill="none"
        height={20}
        viewBox="0 0 20 20"
        width={20}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clipRule="evenodd"
          d="M1.875 1.25H18.125C19.1602 1.25 20 2.08984 20 3.125V16.875C20 17.9102 19.1602 18.75 18.125 18.75H1.875C0.839844 18.75 0 17.9102 0 16.875V3.125C0 2.08984 0.839844 1.25 1.875 1.25ZM17.8906 16.875C18.0195 16.875 18.125 16.7695 18.125 16.6406V7.5H1.875V16.6406C1.875 16.7695 1.98047 16.875 2.10938 16.875H17.8906Z"
          fill="black"
          fillRule="evenodd"
        />
        <mask
          height="18"
          id="mask0_277_1709"
          maskUnits="userSpaceOnUse"
          width="20"
          x="0"
          y="1"
        >
          <path
            clipRule="evenodd"
            d="M1.875 1.25H18.125C19.1602 1.25 20 2.08984 20 3.125V16.875C20 17.9102 19.1602 18.75 18.125 18.75H1.875C0.839844 18.75 0 17.9102 0 16.875V3.125C0 2.08984 0.839844 1.25 1.875 1.25ZM17.8906 16.875C18.0195 16.875 18.125 16.7695 18.125 16.6406V7.5H1.875V16.6406C1.875 16.7695 1.98047 16.875 2.10938 16.875H17.8906Z"
            fill="white"
            fillRule="evenodd"
          />
        </mask>
        <g
          mask="url(#mask0_277_1709)"
        >
          <path
            clipRule="evenodd"
            d="M0 0H20V20H0V0Z"
            fill="#72767b"
            fillRule="evenodd"
          />
        </g>
      </svg>
    </span>
    <span
      className="sc-crXcEl hxLnNX"
    >
      Item 1
    </span>
  </a>
</li>
`;
